.golangci.yml

```diff
@@ -1,98 +1,113 @@
+version: "2"
 linters:
   enable:
-    - depguard # Checks for dependencies that should not be (re)introduced. See "linter-settings" for further details.
     - copyloopvar # Checks for loop variable copies in Go 1.22+
-    - gofmt
-    - goimports
+    - depguard # Checks for dependencies that should not be (re)introduced. See "settings" for further details.
+    - dupword # Checks for duplicate words in the source code
     - gosec
-    - ineffassign
+    - revive
     - misspell
     - nolintlint
-    - revive
-    - staticcheck
-    - tenv # Detects using os.Setenv instead of t.Setenv since Go 1.17
     - unconvert
-    - unused
-    - govet
-    - dupword # Checks for duplicate words in the source code
   disable:
     - errcheck
-
+  settings:
+    depguard:
+      rules:
+        main:
+          deny:
+            - pkg: github.com/opencontainers/runc
+              desc: We don't want to depend on runc (libcontainer), unless there is no other option; see https://github.com/opencontainers/runc/issues/3028.
+    forbidigo:
+      forbid:
+        - pkg: ^regexp$
+          msg: Use internal/lazyregexp.New instead.
+    gosec:
+      # The following issues surfaced when `gosec` linter
+      # was enabled. They are temporarily excluded to unblock
+      # the existing workflow, but still to be addressed by
+      # future works.
+      excludes:
+        - G204
+        - G305
+        - G306
+        - G402
+        - G404
+        - G115
+        - G103
+        - G104
+        - G301
+        - G302
+        - G304
+    staticcheck:
+      checks:
+        - all
+        - -QF1008 # Excludes QF1008 from staticcheck
+        - -ST1000
+        - -ST1020
+        - -ST1021
```

Comment:
Same for these; `# Excludes QF1000 from staticcheck` describes what it does, but doesn't describe "why" / "motivation". Also see https://github.com/moby/moby/blob/cee6f64ac7c475f80a44c28d76ad72f02b910146/.golangci.yml#L152-L158